XMonad/Hooks/DynamicLog.hs

```diff
@@ -746,6 +766,192 @@ xmobarStripTags tags = strip [] where
 filterOutWsPP :: [WorkspaceId] -> PP -> PP
 filterOutWsPP ws pp = pp { ppSort = (. filterOutWs ws) <$> ppSort pp }
 
+
+-- $internals
+-- Internally, the status bars are managed through the 'StatusBarConfig' datatype that provides a
+-- convenient abstraction over what a status bar is and how to manage it; and the 'makeStatusBar'
+-- which and binds it to an 'XConfig'.
+-- Instead of directly modifying the config, the functions 'statusBarProp' and 'statusBar', create an
+-- appropriate 'StatusBarConfig' and call 'makeStatusBar' on it. This enables the users to have more
+-- flexibility and control over the status bars, without having do the plumbing themselves.
+--
+-- For example, if you want to use property logging as provided by 'statusBarProp', but you want to
+-- customize the action executed to toggle the struts, you could do the following:
+--
+-- > basePropSB = statusBarPropConfig "xmobar" myPP -- default key: mod-b
+-- > mySB = withAction (sendMessage ToggleGaps *> sendMessage ToggleStruts) <$> basePropSB
+-- > main = xmonad =<< (mySB >>= flip makeStatusBar) myConf
```

Comment:
You're right

Reply:
It should be fixed now

Reply:
Because the flip is gone it may now be nice to incorporate the config into the status bar function right away and write something like 

``` haskell
mySB = makeStatusBar myConf . withAction (sendMessage ToggleStruts) =<< basePropSB
-- or, if that makes things clearer
mySB = makeStatusBar myConf =<< (withAction (sendMessage ToggleStruts) <$> basePropSB)
```

Reply:
Don't you think that might confuse users on how to compose it with other functions?

Reply:
To elaborate: I typically want to have a set of "modifiers" to my base config, that I can simply toggle on and off, so having something that composes with `.` would be convenient.

Reply:
After playing with my config a bit, I think your suggestion is also fine.  I updated the docs accordingly